server/src/main/java/org/candlepin/dto/api/v1/PoolToSubscriptionTranslator.java

```diff
@@ -74,13 +74,13 @@ public SubscriptionDTO populate(
             throw new IllegalArgumentException("dest is null");
         }
 
-        dest.id(source.getId())
+        dest.id(source.getSubscriptionId())
             .quantity(source.getQuantity())
```

Comment:
Unfortunately, it is not as simple as changing the source on the translator from `Subscription` to `Pool`.
When the fabricate method creates a new Subscription object, it uses this constructor: `public Subscription populate(Pool source, ProductCurator productCurator)` which itself does A LOT of work that involves logic, and the ProductCurator. For example, we can't just assign the quantity, but calculate it first with `Subscription.setQuantityFromPool`.

We should move all that logic from the `Subscription.populate(source, productCurator)` method here because it is basically a translator method hidden within the Subscription class, and entirely remove the following methods, because they are will no longer be needed:
`Subscription.Subscription(Pool source, ProductCurator productCurator)`
`Subscription.populate(Pool source, ProductCurator productCurator)`
`Subscription.setQuantityFromPool(Pool pool)`
`CandlepinPoolManager.fabricateSubscriptionFromPool(Pool pool)`
`PoolManager.fabricateSubscriptionFromPool(Pool pool)`

Reply:
Updated the changes as per the `Subscription.populate` method. Also removed the not required methods.